docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -107,18 +107,22 @@ type Asset = {
   hash: string;
   key: string;
   contentType: string;
+  fileExtension?: string;
   url: string;
 }
 ```
   * `id`: The ID MUST uniquely specify the manifest.
   * `createdAt`: The date and time at which the update was created is essential as the client library selects the most recent update (subject to any constraints supplied by the `expo-manifest-filters` header). The datetime should be formatted according to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601).
   * `runtimeVersion`: Can be any string defined by the developer. It stipulates what native code setup is required to run the associated update.
-  * `launchAsset`: A special asset that is the entry point of the application code.
+  * `launchAsset`: A special asset that is the entry point of the application code. The `fileExtension` field will be ignored for this asset and SHOULD be omitted.
   * `assets`: An array of assets used by the update bundle, such as JavaScript, pictures, and fonts. All assets (including the `launchAsset`) should be downloaded to disk before executing the the update, and a mapping of asset `key`s to locations on disk should be provided to application code.
   * Properties of each asset object:
     * `hash`: SHA-256 hash of the file to guarantee integrity.
     * `key`: Key used to reference this asset from the update's application code. This key, for example, may be generated by a separate build step that processes the application code, such as a bundler.
     * `contentType`: The MIME type of the file as defined by [RFC 2045](https://tools.ietf.org/html/rfc2045). e.g. `application/javascript`, `image/jpeg`.
+    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. `.jpeg`. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end. 
+    A conforming server MUST return a 400 if a file extension without a `.` prefix is uploaded.
```

Comment:
Remove this. A server doesn't necessarily receive uploads. It can just serve updates without receiving them.

It's also valid for a server that receives uploads to get an extension without a . and to add the . in itself. Doesn't matter as long as the server responds with a . which is already described earlier.

Reply:
true.